api/managementpb/checks.proto

```diff
@@ -77,8 +110,66 @@ message ChangeSecurityChecksRequest {
 
 message ChangeSecurityChecksResponse {}
 
+message ListFailedServicesRequest {}
+
+message ListFailedServicesResponse {
+  repeated CheckResultSummary result = 1;
+}
+
+message GetFailedChecksRequest {
+  string service_id = 1;
+  PageParams page_params = 2;
+}
+
+message GetFailedChecksResponse {
+  repeated CheckResult results = 1;
+  PageTotals page_totals = 2;
+}
+
+message ToggleCheckAlertRequest {
+  // Alert ID of the check result.
+  string alert_id = 1;
+  // Silence the check result if set to true, unsilence otherwise.
+  bool silence = 2;
+}
+
+message ToggleCheckAlertResponse {}
+
 // SecurityChecks service provides public Management API methods for Security Checks Service.
 service SecurityChecks {
+  // ListFailedServices returns a list of services with failed checks.
+  rpc ListFailedServices(ListFailedServicesRequest) returns (ListFailedServicesResponse) {
+    option (google.api.http) = {
+      post: "/v1/management/SecurityChecks/ListFailedServices"
+      body: "*"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "List Failed Services"
+      description: "Returns a list of services with failed checks and a summary of check results."
+    };
+  }
+  // GetFailedChecks returns the checks result for a given service.
+  rpc GetFailedChecks(GetFailedChecksRequest) returns (GetFailedChecksResponse) {
+    option (google.api.http) = {
+      post: "/v1/management/SecurityChecks/FailedChecks"
+      body: "*"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Get Failed Checks"
+      description: "Returns the latest check results for a given service."
+    };
+  }
+  // ToggleCheckAlert allows to switch alerts state for a check result between "silenced" and "unsilenced".
+  rpc ToggleCheckAlert(ToggleCheckAlertRequest) returns (ToggleCheckAlertResponse) {
+    option (google.api.http) = {
+      post: "/v1/management/SecurityChecks/ToggleCheckAlert"
+      body: "*"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Toggle Check Alert"
+      description: "Silence/Unsilence alerts for a specific check result."
+    };
+  }
   // GetSecurityCheckResults returns Security Thread Tool's latest checks results.
   rpc GetSecurityCheckResults(GetSecurityCheckResultsRequest) returns (GetSecurityCheckResultsResponse) {
```

Comment:
As I remember we decided to deprecate this endpoint, right?

Reply:
Yeah, you're right